### ISSUE TITLE
Expose more of TorchVision's backbones via `fastai.vision.models`

### ISSUE THREAD

**Issue by datumbox** [fastai/fastai#3789]
Thanks for your hard work on developing this awesome library!

Currently fastai exposes some of the old backbones of TorchVision such as the ResNet, SqueezeNet, DenseNet and VGG:
https://github.com/fastai/fastai/blob/888a85877f8c3d5890c4ef71db9fe65204414d9e/fastai/vision/models/tvm.py#L1-L4

Unfortunately the file hasn't been modified since late 2020 and as a result it's missing all the new backbones (such as ViT, Swin Transformers, ConvNext, RegNet, EfficientNets, MobileNets etc). It might be worth exposing those modern architectures via fastai.

If that's of interest, then there are a few approaches we could take:
1. We can manually and explicitly add all the new model builder methods in the aforementioned file. 
   - Pros: Clean, simple and will work with static analysis.
   - Cons: We need to manually update this list periodically. Will require defining a minimum TorchVision version in the project dependencies in order to access the newer architectures. Unless we pin the latest TorchVision, users won't be able to access the latest models.
2. We can import everything from `torchvision.models`:
   - Pros: Simple and will work with static analysis. Doesn't require defining specific versions of TorchVision.
   - Cons: Importing `*` can be problematic as it will bring in submodules and methods that are not model builders (things like the `get_weight()` method for example).
3. We can create a dynamic solution using TorchVision's new [Model Registration API](https://pytorch.org/blog/easily-list-and-initialize-models-with-new-apis-in-torchvision/):
   - Pros: Doesn't require updates and exposes automatically the right methods.
   - Cons: It's more complex solution. The dynamic approach might make some static analyzers struggle.

There might be other options I missed or hybrid approaches. I would love to hear your thoughts on whether this would be of interest and what option you would favour. Happy to send a PR if this is a direction you would like to take.

Finally note that the Model Registration is currently considered Beta. We've intentionally marked it as such to ensure that we can act upon the feedback of our partners and users. We are currently collecting feedback at https://github.com/pytorch/vision/issues/6365 and hoping to promote to stable before TorchVision v0.14. If you have any feedback or feature requests, we would be happy to discuss more.

cc @jph00

**Comment by jph00** [comment]
I think it's useful to have an easy way to import all models. Might you be open to adding a submodule in torchvision which folks could import to just get the model builders with an `import *`? Or alternatively, define `__all__` in `torchvision.models` to only include the model builders?

One thing to consider is how to handle transfer learning with the new models. So far we've been manually creating a metadata dict with info about where to cut the head to add a new transfer learning head, and what normalization statistics were used in training. However the timm library (which we also support in fastai) has these things available directly as part of the library for each model, so we can simply grab the info from there. That would make it easier for us to support all the new models. In addition, timm also has metadata which tell us which layers to use when creating a dynamic unet -- currently for torchvision models we have to figure this out for ourselves; that would be another nice thing to add to torchvision IMO.

**Comment by datumbox** [comment]
Thanks a lot for coming back to me @jph00.

> Might you be open to adding a submodule in torchvision which folks could import to just get the model builders with an import *? Or alternatively, define __all__ in torchvision.models to only include the model builders?

Currently we don't define an `__all__` in [`torchvision/models/__init__.py`](https://github.com/pytorch/vision/blob/main/torchvision/models/__init__.py) and so everything that is public is automatically exposed:
```python
# Import all backbone architecture methods + their weight enums:
from .alexnet import *
# ... more backbone imports go here...

# Import submodules:
from . import detection, optical_flow, quantization, segmentation, video  

# Import other non-model builder methods:
from ._api import get_model, get_model_weights, get_weight, list_models 
```

As you can see not all publicly exposed items are model builders. We include model classes (like `AlexNet`), weight enums (like `AlexNet_Weights`), model utility methods (like `interpolate_embeddings()` for ViT), other API methods (like `get_weight()`) and even submodules (like `detection`). This is one of the primary reasons we are introducing the registration API, so that one can easily get just the backbones with a single method call.

If you are not inclined to adopt the new API, then perhaps you would consider a workaround like the following? We can add tests on our side to ensure it will always return the expected results for you:
```python
from torchvision.models import *
for name in {"detection", "segmentation", "get_weight", "list_models", ...}:
    __builtins__.globals().pop(name)
```

I'm happy to discuss alternatives. Just wanted to provide a few more details on the limitations and current situation on TorchVision.

> One thing to consider is how to handle transfer learning with the new models. So far we've been manually creating a metadata dict with info about where to cut the head to add a new transfer learning head, and what normalization statistics were used in training

That makes a lot of sense. I would love to get more details on exactly the type of meta-data you would like to store. Perhaps you could provide a few code references on how exactly you store this info? 

One potential option is to use the meta-data provided by the [Multi-weight support API](https://pytorch.org/blog/introducing-torchvision-new-multi-weight-support-api/). This was introduced in v0.13 and allows us to store things like meta-data and the preprocessing transforms along with the weights to facilitate transfer learning. I would like to understand more about your use-case to see if that is the right place to put this information. For reference [here](https://github.com/pytorch/vision/blob/main/torchvision/models/alexnet.py#L55-L74) is an example of the things we currently store.

**Comment by datumbox** [comment]
@jph00 I took a stab at option 2 at PR #3791. Let me know your thoughts.
